src/test/java/uk/gov/digital/ho/pttg/AuditEntryJpaRepositoryTest.java

```diff
@@ -277,6 +277,55 @@ public void findArchivedResults_multipleResults_correctDatesReturned() {
         assertThat(results).containsExactly(archivedResult2);
     }
 
+    @Test
+    public void getAllCorrelationIds_givenEventType_returnCorrelationIds() {
+        String correlationId = "some correlation id";
+        repository.save(createAudit(correlationId, INCOME_PROVING_FINANCIAL_STATUS_RESPONSE));
+
+        List<String> correlationIds = repository.getAllCorrelationIds(singletonList(INCOME_PROVING_FINANCIAL_STATUS_RESPONSE));
+        assertThat(correlationIds)
+                .contains(correlationId);
+    }
+
+    @Test
+    public void getAllCorrelationIds_givenEventType_returnOnlyCorrelationIdsForEventType() {
+        String correlationId1 = "some correlation id";
+        String correlationId2 = "some other correlation id";
+        String correlationId3 = "yet some other correlation id";
+        repository.save(createAudit(correlationId1, INCOME_PROVING_INCOME_CHECK_REQUEST));
+        repository.save(createAudit(correlationId2, INCOME_PROVING_INCOME_CHECK_REQUEST));
+        repository.save(createAudit(correlationId3, INCOME_PROVING_FINANCIAL_STATUS_RESPONSE));
+
+        List<String> correlationIds = repository.getAllCorrelationIds(singletonList(INCOME_PROVING_INCOME_CHECK_REQUEST));
+        assertThat(correlationIds)
+                .contains(correlationId1, correlationId2);
+    }
+
+    @Test
+    public void getAllCorrelationIds_multipleEventTypes_returnCorrelationIdsForEventTypes() {
+        String correlationId1 = "some correlation id";
+        String correlationId2 = "some other correlation id";
+        String correlationId3 = "yet some other correlation id";
+        repository.save(createAudit(correlationId1, INCOME_PROVING_INCOME_CHECK_REQUEST));
+        repository.save(createAudit(correlationId2, DWP_BENEFIT_REQUEST));
+        repository.save(createAudit(correlationId3, INCOME_PROVING_FINANCIAL_STATUS_RESPONSE));
+
+        List<String> correlationIds = repository.getAllCorrelationIds(Arrays.asList(INCOME_PROVING_INCOME_CHECK_REQUEST, DWP_BENEFIT_REQUEST));
+        assertThat(correlationIds)
+                .contains(correlationId1, correlationId2);
+    }
+
+    @Test
+    public void getAllCorrelationIds_multipleEntriesPerCorrelationId_returnDistinctIds() {
+        String correlationId = "some correlation id";
+        repository.save(createAudit(correlationId, INCOME_PROVING_INCOME_CHECK_REQUEST));
+        repository.save(createAudit(correlationId, INCOME_PROVING_INCOME_CHECK_REQUEST));
```

Comment:
Is this the same primary key, so will only end up with a single audit entry? I think we have idempotent writes, so you just need different event types for the same correlation id.